python/bindings.cpp

```diff
@@ -45,9 +31,385 @@ PYBIND11_MODULE(pylasr, m) {
     m.def("get_total_ram", &getTotalRAM,
           "Get the total RAM in bytes");
 
-    // Export Pipeline class
-    py::class_<Pipeline>(m, "Pipeline")
-        .def(py::init<>())
-        .def("need_buffer", &Pipeline::need_buffer)
-        .def("clear", &Pipeline::clear);
-} 
+    // LAS utility functions
+    m.def("las_filter_usage", &api::lasfilterusage,
+          "Display LASlib filter usage information");
+
+    m.def("las_transform_usage", &api::lastransformusage,
+          "Display LASlib transform usage information");
+
+    m.def("is_indexed", &api::is_indexed,
+          "Check if a LAS file is spatially indexed",
+          py::arg("file"));
+
+    // Pipeline info structure
+    py::class_<api::PipelineInfo>(m, "PipelineInfo")
+        .def_readwrite("streamable", &api::PipelineInfo::streamable)
+        .def_readwrite("read_points", &api::PipelineInfo::read_points)
+        .def_readwrite("buffer", &api::PipelineInfo::buffer)
+        .def_readwrite("parallelizable", &api::PipelineInfo::parallelizable)
+        .def_readwrite("parallelized", &api::PipelineInfo::parallelized)
+        .def_readwrite("use_rcapi", &api::PipelineInfo::use_rcapi);
+
+    // Core API functions
+    m.def("execute", &api::execute,
+          py::call_guard<py::gil_scoped_release>(),
+          "Execute a pipeline from a JSON configuration file",
+          py::arg("config_file"),
+          py::arg("async_communication_file") = "");
+
+    m.def("pipeline_info", &api::pipeline_info,
+          "Get pipeline information from a JSON configuration file",
+          py::arg("config_file"));
+
+    // Export Stage class from the C++ API
+    py::class_<api::Stage>(m, "Stage")
```

Comment:
In R I'm not exposing the classes `Stage` and `Pipeline`. This is a design choice. I could have chosen another strategy by exposing the `Pipeline` class. However I'm wondering if there is any interest at exposing the `Stage`. A stage is always wrapped into a `Pipeline`. The only class useful for a user is `Pipeline`. Just a through.